stretch-assignment/index.js

```diff
@@ -0,0 +1,34 @@
+const blocks = document.querySelectorAll('.block');
+
+let addX = (x) => x + 1;
+let subX = (x) => x - 1;
+
+blocks.forEach(block => {
+  let x = 0;
+  let blockID;
+  let handleX;
+
+  block.addEventListener('mouseover', (event) => {
+    handleX = addX;
+    if (!blockID) {
+      blockID = setInterval(() => {
+        x = handleX(x);
+        event.target.style.transform = `translateX(${x}px)`;
+        if (x === 0) {
+          clearInterval(blockID);
+          blockID = null;
+        }
+      }, 10);
+    }
+  });
+
+  block.addEventListener('mouseout', (event) => { handleX = subX; });
+
+  block.addEventListener('click', (event) => {
+    blocks.forEach(el => {
+      if (el !== event.target) el.style.order = 0;
+    });
+
+    event.target.style.order = -1;
+  });
+});
```

Comment:
very clever way of handling this, going off of forEach-scoped variables for both the mouseover and mouseout events is the perfect way to handle the behavior we're looking for, especially when re-entering a block while it's transitioning back to its starting position!